File: Zadanie1/Zadanie1/CurrencyOperations.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Zadanie1
{
    public class CurrencyOperations
    {
        public decimal DollarToEuro(decimal Dolars) => Dolars * 1.01m;
        public decimal EuroToDollar(decimal Euros) => Euros * 0.99m;
        public decimal DollarToPLN(decimal Dollar) => Dollar * 4.84m;
        public decimal PLNToDollar(decimal PLN) => PLN * 0.21m;
        public decimal EuroToPLN(decimal Euros) => Euros * 4.78m;
        public decimal PLNToEuro(decimal PLN) => PLN * 0.21m;

        // return in dollars
        public decimal AddDollarsAndEuro(decimal dollars, decimal euro) => dollars + EuroToDollar(euro);

        // return in euro
        public decimal AddEuroAndDollars(decimal euro, decimal dollars) => euro + DollarToEuro(dollars);

        // return in PLN
        public decimal AddDollarsAndEuroToPLN(decimal dollars, decimal euro) => DollarToPLN(dollars) + EuroToPLN(euro);

        // return in PLN
        public decimal SubEuroFromDollarsToPLN(decimal dollars, decimal euro) => dollars + EuroToDollar(euro);
    }
}

File: PoprzednieZadania/Zadanie3/src/main/java/pl/pjwstk/s18946/CurrencyOperations.java
package pl.pjwstk.s18946;

public class CurrencyOperations {

    public double dollarToEuro(double dollars) {
        return dollars * 1.01;
    }

    public double euroToDollar(double euros) {
        return euros * 0.99;
    }

    public double dollarToPLN(double dollars) {
        return dollars * 4.84;
    }

    public double plnToDollar(double pln) {
        return pln * 0.21;
    }

    public double euroToPLN(double euros) {
        return euros * 4.78;
    }

    public double plnToEuro(double pln) {
        return pln * 0.21;
    }

    public double addDollarsAndEuro(double dollars, double euro) {
        return dollars + euroToDollar(euro);
    }

    public double addEuroAndDollars(double euro, double dollars) {
        return euro + dollarToEuro(dollars);
    }

    public double addDollarsAndEuroToPLN(double dollars, double euro) {
        return dollarToPLN(dollars) + euroToPLN(euro);
    }

    public double subEuroFromDollarsToPLN(double dollars, double euro) {
        return dollars + euroToDollar(euro);
    }
}
